drivers/watchdog/wdt_mcux_imx_rtwdog.c

```diff
@@ -0,0 +1,222 @@
+	struct mcux_rtwdog_data *data = dev->data;
+	RTWDOG_Type *base = config->base;
+
+	RTWDOG_Deinit(base);
+	data->timeout_valid = false;
+	LOG_DBG("Disabled the watchdog");
+
+	return 0;
+}
+
+static int mcux_rtwdog_install_timeout(const struct device *dev, const struct wdt_timeout_cfg *cfg)
+{
+	const struct mcux_rtwdog_config *config = dev->config;
+	struct mcux_rtwdog_data *data = dev->data;
+	uint32_t clock_freq;
+	uint32_t timeoutCycle;
+	uint32_t div;
+
+	if (data->timeout_valid) {
+		LOG_ERR("No more timeouts can be installed");
+		return -ENOMEM;
+	}
+
+	RTWDOG_GetDefaultConfig(&data->wdog_config);
+
+#if DT_NODE_HAS_PROP(DT_INST_PHANDLE(0, clocks), clock_frequency)
+	clock_freq = config->clock_frequency;
+#else  /* !DT_NODE_HAS_PROP(DT_INST_PHANDLE(0, clocks), clock_frequency) */
+	if (!device_is_ready(config->clock_dev)) {
+		LOG_ERR("clock control device not ready");
+		return -ENODEV;
+	}
+
+	if (clock_control_get_rate(config->clock_dev, config->clock_subsys, &clock_freq)) {
+		return -EINVAL;
+	}
+#endif /* !DT_NODE_HAS_PROP(DT_INST_PHANDLE(0, clocks), clock_frequency) */
+
+	div = config->clk_divider == kRTWDOG_ClockPrescalerDivide1 ? 1U : 256U;
+
+	data->wdog_config.clockSource = config->clk_source;
+	data->wdog_config.prescaler = config->clk_divider;
+
+	timeoutCycle = MSEC_TO_RTWDOG_TICKS(clock_freq, div, cfg->window.max);
+	if (timeoutCycle > U16_MAX) {
+		LOG_ERR("Invalid window max");
+		return -EINVAL;
+	}
+	data->wdog_config.timeoutValue = timeoutCycle;
+
+	if (cfg->window.min) {
+		data->wdog_config.enableWindowMode = true;
+		data->wdog_config.windowValue =
+			MSEC_TO_RTWDOG_TICKS(clock_freq, div, cfg->window.min);
+	} else {
+		data->wdog_config.enableWindowMode = false;
+		data->wdog_config.windowValue = 0;
+	}
+
+	data->wdog_config.enableInterrupt = cfg->callback != NULL;
+	data->callback = cfg->callback;
+	data->timeout_valid = true;
+
+	return 0;
+}
+
+static int mcux_rtwdog_feed(const struct device *dev, int channel_id)
+{
+	const struct mcux_rtwdog_config *config = dev->config;
+	RTWDOG_Type *base = config->base;
+
+	if (channel_id != 0) {
+		LOG_ERR("Invalid channel id");
+		return -EINVAL;
+	}
+
+	RTWDOG_Refresh(base);
+	LOG_DBG("Fed the watchdog");
+
+	return 0;
+}
+
+static void mcux_rtwdog_isr(const struct device *dev)
+{
+	const struct mcux_rtwdog_config *config = dev->config;
+	struct mcux_rtwdog_data *data = dev->data;
+	RTWDOG_Type *base = config->base;
+	uint32_t flags;
+
+	flags = RTWDOG_GetStatusFlags(base);
+	RTWDOG_ClearStatusFlags(base, flags);
+
+	if (data->callback) {
+		data->callback(dev, 0);
+	}
+}
+
+static int mcux_rtwdog_init(const struct device *dev)
+{
+	const struct mcux_rtwdog_config *config = dev->config;
+	int ret;
+
+	config->irq_config_func(dev);
+
+	ret = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_DEFAULT);
+	if (ret < 0 && ret != -ENOENT) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static const struct wdt_driver_api mcux_rtwdog_api = {
```

Comment:
Drivers need to wrap APIs.

```suggestion
static DEVICE_API(wdt, mcux_rtwdog_api) = {
```